core/test/com/treode/async/misc/private/DirectDequeSpec.scala

```diff
@@ -0,0 +1,125 @@
+package com.treode.async.misc
+
+import org.scalatest.FlatSpec
+
+class DirectDequeSpec extends FlatSpec {
+	
+	"DirectDeque" should "add one integer to the end of the deque" in {
+		val deque = new DirectDeque[Int]
+		// enqueue one element
+		deque.enqueue(1)
+		assert (!deque.isEmpty())
+		assert (deque.size() == 1)
+	}
+
+	it should "be able to add more than the deque initial capacity" in {
+		val deque = new DirectDeque[Int]
+		// enqueue 17 elements
+		for (i <- 1 to 17) {
```

Comment:
We know that `17` derives from initSize, but it may look like a magic number to the next person through. Make the initSize a parameter to the constructor. Then it's clear that the limit in this loop relates to the initSize. Also, that lets you lower the initSize for testing.